### PR TITLE
Document rich relationships

### PR DESCRIPTION
**Pull request by wheresrhys**
## Why?
We will soon be launching the ability to annotate dependency relationships. We support doing this in RUNBOOK.md, but the only documentation in existence is quite abstract.

## What has changed?
Adds some docs that concentrate on the specific first example, while introducing some more general ideas about the format

Also fixes local app running (there was a docker problem)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -73,11 +73,13 @@ run-local-stream-container:
 	# first check if an exited container blocks (and remove it)
 	# then run the container with the kinesalite kinesis emulator
 	# see https://docs.docker.com/engine/reference/commandline/ps/
+	# docker ps -a | grep localstreams | awk '{print $1}' | xargs docker rm -f; \
```
Comment (wheresrhys) [inline Makefile]:
```suggestion
```

File: docs/dependency-details.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Adding details to dependency relationships
+
+We have recently added support to adding additional information to dependency relationships.
```
Comment (wheresrhys) [inline docs/dependency-details.md]:
```suggestion
We have recently added support for adding additional information to dependency relationships.
```

File: docs/dependency-details.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+In RUNBOOK.md dependencies are specified as follows:
+
+```md
+## Dependencies
+
+-   biz-ops-api
+-   biz-ops-schema
+```
+
+They support adding 2 kinds of additional details:
```
Comment (wheresrhys) [inline docs/dependency-details.md]:
```suggestion
Now you can add 2 kinds of additional details:
```

File: docs/dependency-details.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+## Dependencies
+
+-   biz-ops-api
+-   biz-ops-schema
+```
+
+Now you can add 2 kinds of additional details:
+
+-   resiliencePatterns: one or more specific architectural patterns that can make a dependency more resilient. Choose from the list: Cache, DurableCache, GracefulDegradation, Replay, ManualFallback
+-   resilienceDetails: A free text field to explain more about how resilience is implemented
```
Comment (wheresrhys) [inline docs/dependency-details.md]:
```suggestion
-   resilienceDetails: A free text field to explain more about how resilience is implemented. This should be entered on a single line
```

File: docs/dependency-details.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+## Dependencies
+
+-   biz-ops-api
+-   biz-ops-schema
+```
+
+Now you can add 2 kinds of additional details:
+
+-   resiliencePatterns: one or more specific architectural patterns that can make a dependency more resilient. Choose from the list: Cache, DurableCache, GracefulDegradation, Replay, ManualFallback
+-   resilienceDetails: A free text field to explain more about how resilience is implemented. This should be entered on a single line
```
Comment (wheresrhys) [inline docs/dependency-details.md]:
```suggestion
-   resilienceDetails: A free text field to explain more about how resilience is implemented.
```

File: docs/dependency-details.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+-   biz-ops-api
+    resiliencePatterns: Cache, Replay
+    resilienceDetails: Calls to Biz Ops are held in an in memory cache. The app consumes a kinesis stream, which replays events that cause the application to error.
+-   biz-ops-schema
+```
+
+## More generally
+
+This is the first example we've added to Biz Ops of adding properties directly onto relationships. This is a powerful addition to the graph model which you shoudl expect to see more of. The general pattern for adding these in RUNBOOK.md is
```
Comment (wheresrhys) [inline docs/dependency-details.md]:
```suggestion
This is the first example we've added to Biz Ops of adding properties directly onto relationships. This is a powerful addition to the graph model which you should expect to see more of. The general pattern for adding these in RUNBOOK.md is
```

File: docs/dependency-details.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+In RUNBOOK.md dependencies are specified as follows:
+
+```md
+## Dependencies
+
+-   biz-ops-api
+-   biz-ops-schema
+```
+
+Now you can add 2 kinds of additional details:
```
Comment (nl-ria) [inline docs/dependency-details.md]:
```suggestion
You can also add 2 kinds of additional details:
```

File: docs/dependency-details.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+-   resiliencePatterns: one or more specific architectural patterns that can make a dependency more resilient. Choose from the list: Cache, DurableCache, GracefulDegradation, Replay, ManualFallback
```
Comment (nl-ria) [inline docs/dependency-details.md]:
```suggestion
-   `resiliencePatterns`: one or more specific architectural patterns that can make a dependency more resilient. Choose from the list: `Cache`, `DurableCache`, `GracefulDegradation`, `Replay`, `ManualFallback`
```

Comment (nl-ria) [reply docs/dependency-details.md]: Wondering if we could add a link to docs which describe what each of these means?

Comment (wheresrhys) [reply docs/dependency-details.md]: good idea

File: lambdas/ingester/src/docsPageHandler.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ const template = require('./templates/docs-page');
 
 const pages = {
 	quickstart: require('../../../docs/quickstart.md'), // eslint-disable-line global-require
+	'dependency-details': require('../../../docs/dependency-details.md'), // eslint-disable-line global-require
```
[the remark this answers is not in the document]
Comment (wheresrhys) [reply lambdas/ingester/src/docsPageHandler.js]: I will add disable start/end statements before and after this object

File: docs/dependency-details.md (hunk 8 above)
Comment (wheresrhys) [reply docs/dependency-details.md]: I'll add a link once the schema is merged to master

**Pull request merged**